### PR TITLE
Chat: display excluded @-files in UI

### PR DESCRIPTION
**Pull request by abeatrix**
image

Update context list to match [figma design ](https://www.figma.com/file/jEsnzgsf0hNuJbqB9pKaGm/VS-Code---Cody---Commands-up-front-and-ChatGPT%2B%2B?type=design&node-id=1826-40270&mode=design&t=uR9EdiEEMOLHwOsN-0)

## Test plan

Expected behavior:

- @-mentioned items that are excluded should show up in the UI

image

- excluded enhanced context will not show up in UI
- no sparkle icon next to context for non-enhanced context items

image
Ex

- context mixed with enhanced context items will display sparkle next to Context:

image

### CODE REVIEW

File: vscode/webviews/chat/components/EnhancedContext.tsx [hunk 1]
```diff
@@ -51,10 +64,13 @@ export const EnhancedContext: React.FunctionComponent<{
...
         0
     )
-    const fileCount = new Set(contextFiles.map(file => file.uri.toString())).size
+    const fileCount = new Set(usedContext.map(file => file.uri.toString())).size
     const lines = `${lineCount} line${lineCount > 1 ? 's' : ''}`
     const files = `${fileCount} file${fileCount > 1 ? 's' : ''}`
-    const title = lineCount ? `${lines} from ${files}` : `${files}`
+    let title = lineCount ? `${lines} from ${files}` : `${files}`
+    if (excludedAtContext.length) {
+        title = `${title} - ⚠️ ${excludedAtContext.length} mentiones excluded`
```
Comment (toolmantim) [inline vscode/webviews/chat/components/EnhancedContext.tsx:72]: s/mentiones/mentions

File: vscode/webviews/Components/FileLink.tsx [hunk 2]
```diff
@@ -32,13 +35,34 @@ export const FileLink: React.FunctionComponent<FileLinkProps> = ({
         )
     }
 
-    const pathToDisplay = `@${displayPath(uri)}`
+    const pathToDisplay = `${displayPath(uri)}`
     const pathWithRange = range ? `${pathToDisplay}:${displayLineRange(range)}` : pathToDisplay
     const tooltip = source ? `${pathWithRange} included via ${source}` : pathWithRange
     const { href, target } = webviewOpenURIForContextItem({ uri, range })
+    const warning = 'This file is excluded due to token limit reached'
```
Comment (toolmantim) [inline vscode/webviews/Components/FileLink.tsx:42]: "Excluded due to context window limit"

File: vscode/webviews/chat/components/EnhancedContext.tsx [hunk 3]
```diff
@@ -30,17 +31,29 @@ export const EnhancedContext: React.FunctionComponent<{
...
     // NOTE: sparkle should only be added to messages that use enhanced context.
     // NOTE: Core chat commands (e.g. /explain and /smell) use local context only.
     // Check if the filteredFiles only contain local context (non-enhanced context).
-    const localContextType = ['user', 'selection', 'terminal', 'editor']
-    const localContextOnly = contextFiles.every(file => localContextType.includes(file.type))
+    const localContextType = ['user', 'uri', 'selection', 'terminal', 'editor']
+    const localContextOnly = usedContext.every(file =>
+        localContextType.includes(file.source || file.type)
+    )
     const sparkle = localContextOnly ? '' : '✨ '
```
Comment (toolmantim) [inline vscode/webviews/chat/components/EnhancedContext.tsx:52]: We can ditch this ✨ emoji in the details title now

File: vscode/webviews/Components/FileLink.tsx [hunk 4]
```diff
@@ -32,13 +35,34 @@ export const FileLink: React.FunctionComponent<FileLinkProps> = ({
...
+            </a>
+        </span>
     )
 }
+
+function getIconByFileSource(source?: string): string {
+    switch (source) {
+        case 'user':
+            return 'mention'
+        case 'uri':
```
Comment (toolmantim) [inline vscode/webviews/Components/FileLink.tsx:63]: Is this @'mentioned URIs? In which case I think we still want to use the same mention icon.

**Review by toolmantim — COMMENTED** [review]
This is great!

A few small things inline, and also…

* Icon & text wraps on small screens:

CleanShot 2024-03-26 at 00 14 38@2x

* We should update the tooltips with this new design.
** Icon’s title: "Included via Enhanced Context (Search)", "Included via Enhanced Context (Enterprise Search)", "Included via Enhanced Context (Embeddings)"
** File link's title: can just be the full file path (drop "included via")

File: vscode/webviews/Components/FileLink.tsx (hunk 4 above)
Comment (abeatrix) [reply vscode/webviews/Components/FileLink.tsx:63]: In your design the `@mentioned URIs` uses the link icon. Do you want it to use the same @ icon?
image

From

image

to

image

**Comment by abeatrix** [comment]
@toolmantim Applied your feedback to the PR with the following changes:

- [x] We can ditch this ✨ emoji in the details title now
- [x] Icon & text wraps on small screens:
image

- [x] @-mention links use @-mention icon
image


- [x] We should update the tooltips with this new design
image

image

image

- [x] Updated exclude warning to "Excluded due to context window limit" on excluded icon hover:
image

**Review by toolmantim — APPROVED** [review]
Awesome!

**Review by abeatrix — COMMENTED** [review]
Comments inline, mostly for the styling on FileLink. 
I think this is a great start and we can refine the styles more in follow-up?

**Pull request merged**